actions/docker-image-patch/action.yml

```diff
@@ -0,0 +1,192 @@
+    required: true
+  copy-paths:
+    description: |
+      List of file paths to update in the Docker image.
+
+      Each path should be in the format `path/to/file:destination/path`.
+      The colon (:) separates the source path from the destination path.
+      The destination path should be absolute. Can be a file or directory.
+
+      Examples:
+        copy-paths: |
+          # Copy all files from the local directory to the Docker image
+          bin/:/usr/local/bin/
+          # Copy a single file from the local directory to the Docker image
+          config.yaml:/app/config.yaml
+    required: false
+  docker-envs:
+    description: |
+      List of environment variables to set in the Docker image.
+
+      Each variable should be in the format `KEY=VALUE`.
+
+      Examples:
+        docker-envs: |
+          ENV_VAR_1=value1
+          ENV_VAR_2=value2
+    required: false
+  docker-image-tag-src:
+    description: |
+      Source Docker image tag.
+
+      Examples:
+        latest
+        1.0.0
+        1.0.0-rc.1
+    required: true
+  docker-image-tag-dst:
+    description: |
+      Destination Docker image tag.
+
+      Examples:
+        latest
+        1.0.0
+        1.0.0-rc.1
+    required: true
+  docker-registry-url-dst:
+    required: true
+    description: |
+      Hostname for the destination Docker image registry.
+
+      Currently only supports public and private ECR.
+
+      Examples:
+        public.ecr.aws
+        <account-id>.dkr.ecr.<region>.amazonaws.com
+  docker-registry-url-src:
+    required: true
+    description: |
+      Hostname for the source Docker image registry.
+
+      Currently only supports public and private ECR.
+
+      Examples:
+        <account-id>.dkr.ecr.<region>.amazonaws.com
+  docker-repository-name-dst:
+    required: true
+    description: |
+      Name of the destination Docker repository excluding hostname. Excludes any tags. Public ECR's will include a registry alias and a forward slash.
+
+      Examples:
+        chainlink/chainlink # Public ECR
+        chainlink/ccip # Public ECR
+        my-repo-name # Private ECR
+  docker-repository-name-src:
+    required: true
+    description: |
+      Name of the source Docker repository excluding hostname. Excludes any tags. Public ECR's will include a registry alias and a forward slash.
+
+      Examples:
+        chainlink/chainlink # Public ECR
+        chainlink/ccip # Public ECR
+        my-repo-name # Private ECR
+
+runs:
+  using: composite
+  steps:
+    - name: Validate inputs
+      shell: bash
+      env:
+        COPY_PATHS: ${{ inputs.copy-paths }}
+        DOCKER_ENVS: ${{ inputs.docker-envs }}
+        DOCKER_REGISTRY_URL_DST: ${{ inputs.docker-registry-url-dst }}
+        DOCKER_REGISTRY_URL_SRC: ${{ inputs.docker-registry-url-src }}
+      run: |
+        echo "::group::Validate inputs"
+        ${GITHUB_ACTION_PATH}/scripts/input-validation.sh
+        echo "::endgroup::"
+
+    - name: Set outputs
```

Comment:
Nit -

It seems unintuitive to "set outputs" before everything else is completed. I see that these values are used in subsequent steps, and I also see that this action has no outputs declared.

So maybe this step could be called 'process-inputs' or something?

Reply:
Yeah, it's confusing because there are outputs at the step level, job level, and also the composite action level.